components/director/internal/tenantmapping/http.go

```diff
@@ -0,0 +1,42 @@
+package tenantmapping
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+func parseRequest(req *http.Request) (DataV2, error) {
+	var data DataV2
+	err := json.NewDecoder(req.Body).Decode(&data)
+	if err != nil {
+		if err == io.EOF {
+			return DataV2{}, errors.New("request body is empty")
+		}
+
+		return DataV2{}, errors.Wrap(err, "while decoding request body")
+	}
+
+	defer func() {
+		err := req.Body.Close()
+		if err != nil {
+			log.Error(err)
+		}
+	}()
+
+	if data.Extra == nil {
+		data.Extra = make(map[string]interface{})
+	}
+
+	return data, nil
+}
+
+func respondWithError(writer http.ResponseWriter, err error, wrapperStr string) {
+	wrappedErr := errors.Wrap(err, wrapperStr)
+	log.Error(wrappedErr)
+
+	http.Error(writer, wrappedErr.Error(), http.StatusInternalServerError)
```

Comment:
Is this really always InternalServerError? What if request body is empty? then it is invalid request, right?

Reply:
Will extend the helper method to be able to define the error code.